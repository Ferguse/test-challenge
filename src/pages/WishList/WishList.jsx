import { useMemo } from 'react';
import { useSelector } from 'react-redux';

import { Card } from 'components/Card';
import { getWishList } from 'selectors/selectors';

import './wishList.scss';

export const WishList = () => {
  const wishList = useSelector(getWishList);
  const wishListData = useMemo(() => (
    Object.values(wishList).map((book) => (
      <Card key={book.id} {...{ ...book }} />
    ))
  ), [wishList]);

  return (
    <section className="category bg-white content">
      <div className="container">
        <div className="category-grid">
          {!wishListData.length && 'Wish list is empty'}
          {wishListData}
        </div>
      </div>
    </section>
  );
};
