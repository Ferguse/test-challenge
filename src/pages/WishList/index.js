export { WishList } from './WishList.jsx';
