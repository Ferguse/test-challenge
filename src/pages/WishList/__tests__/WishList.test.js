import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockedStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';

import { WishList } from '../WishList';

describe('Page [WishList]', () => {
  const state = {
    wishList: {},
    categories: {
      entities: {
        bike: {
          test: {
            id: 'test',
            volumeInfo: {},
            category: {},
          },
        }},
    },
    current: {
      category: 'bike',
      book: 'test',
    },
  };

  const store = configureMockedStore([thunk])(state);
  const history = createMemoryHistory({ initialEntries: ['/home'] });

  it('Renders', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router history={history} location={history.location} navigator={history}>
            <WishList />
          </Router>
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
