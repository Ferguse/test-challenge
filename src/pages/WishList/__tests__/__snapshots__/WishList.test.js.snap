// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page [WishList] Renders 1`] = `
<section
  className="category bg-white content"
>
  <div
    className="container"
  >
    <div
      className="category-grid"
    >
      Wish list is empty
    </div>
  </div>
</section>
`;
